File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashtable={}
        for i,n in enumerate(nums):
            if target-nums[i] in hashtable:
                return [hashtable[target-nums[i]],i]
            hashtable[n]=i
        return []

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.java
class Solution {
    public int findMin(int[] nums) {
        int min=nums[0];
        int l=0;
        int r=nums.length-1;
        int mid=0;
        while(l<r){
            mid=(l+r)/2;
            if(nums[r]<nums[mid]){
                l=mid+1;
            }else{
                r=mid;
            }
        }
        return nums[l];
        
    }
}
